src/main/java/seedu/module/model/module/Module.java

```diff
@@ -0,0 +1,88 @@
+package seedu.module.model.module;
+
+import java.util.Objects;
+
+/**
+ * Represents a Module in the ModuleList.
+ */
+public class Module {
+
+    // Identity fields
+    private final String moduleCode;
+    private final String title;
+    private final String description;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Module(String moduleCode, String title, String description) {
+        this.moduleCode = moduleCode;
+        this.title = title;
+        this.description = description;
+    }
+
+    public String getModuleCode() {
+        return moduleCode;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Returns true if both modules of the same name have at least one other identity field that is the same.
+     * This defines a weaker notion of equality between two modules.
+     */
+    public boolean isSameModule(Module otherModule) {
+        if (otherModule == this) {
+            return true;
+        }
+
+        return otherModule != null
```

Comment:
i think the last operator is supposed to be "||" instead of "&&"? if not there is no difference between this and equals method.

Reply:
Good catch. I modified it to only check `moduleCode` for identity equality.